rts-server/src/main.rs

```diff
@@ -65,28 +65,51 @@ pub fn create_user<'a>(
         email,
     };
 
-    diesel::insert_into(users::table)
+    match diesel::insert_into(users::table)
         .values(&new_user)
         .get_result(conn)
-        .expect("Error creating user")
+    {
+        Ok(user) => Ok(user),
+        Err(err) => Err(err.to_string()),
+    }
+}
+
+#[derive(Serialize)]
+#[serde(untagged)]
+enum RegisterResult {
+    Successful,
+    Failed(String),
 }
 
 #[post("/register")]
-async fn register(info: web::Json<RegisterInfo>, state: web::Data<AppState<'_>>) -> impl Responder {
+async fn register(
+    info: web::Json<RegisterInfo>,
+    req: HttpRequest,
+    state: web::Data<AppState<'_>>,
+) -> impl Responder {
     let conn = state.pool.get().expect("Could not connect to the database");
     println!(
         "Register request from {} with password {} and mail {}",
         info.username, info.password, info.email
     );
-    let user = create_user(
+    match create_user(
```

Comment:
You seems to pass the whole struct `RegisterInfo` field by field, consider passing the whole struct to make the function easier to read.